core/embed/trezorhal/optiga/optiga.c

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "optiga.h"
+#include "optiga_commands.h"
+
+int optiga_sign(uint8_t index, const uint8_t *digest, size_t digest_size,
+                uint8_t *signature, size_t max_sig_size, size_t *sig_size) {
+  optiga_result ret =
+      optiga_calc_sign(0xE0F0 + index, digest, digest_size, &signature[2],
+                       max_sig_size - 2, sig_size);
+  if (ret == OPTIGA_ERR_CMD) {
+    uint8_t error_code = 0;
+    optiga_get_error_code(&error_code);
+    return error_code + OPTIGA_COMMAND_ERROR_OFFSET;
+  }
+
+  if (ret != OPTIGA_SUCCESS) {
+    return ret;
+  }
+
+  // Add sequence tag and length.
+  if (*sig_size >= 0x80) {
+    // Length not supported.
+    return OPTIGA_ERR_SIZE;
+  }
+  signature[0] = 0x30;
+  signature[1] = *sig_size;
+  *sig_size += 2;
+  return OPTIGA_SUCCESS;
+}
+
+bool optiga_cert_size(uint8_t index, size_t *cert_size) {
```

Comment:
acceptable range for index?

Reply:
https://github.com/trezor/trezor-firmware/pull/3255/commits/b2badfaa3e1d9aa66b1c5f5dfcc65344d3f81b2e